flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/SharedBuffer.java

```diff
@@ -82,9 +82,12 @@
 
 	private transient Map<K, SharedBufferPage<K, V>> pages;
 
+	private final transient List<SharedBufferEntry<K, V>> prunedEntries;
```

Comment:
Is it necessary for it to be a field? Isn't a local variable sufficient?

Reply:
It is for object reuse. What's your thought?

Reply:
Not sure about the potential advantages. See e.g. this answer on SO: https://stackoverflow.com/a/18371813/4250114 . Plus the local variable is easier to reason about.

Personally I am for the local variable, but if you feel this version is better, will merge it as it is.

Reply:
Make sense, I will change it to local variable.

Reply:
Updated the PR. Thanks a lot for the review. :)